### PR TITLE
Avoid using __import__ and eval

### PR DESCRIPTION
**Pull request by c-bata**
See https://docs.python.org/3/library/functions.html#__import__

```
$ python -V
Python 3.7.2
$ gunicorn examples.echo:app
[2019-05-20 23:09:16 +0900] [26697] [INFO] Starting gunicorn 19.9.0
[2019-05-20 23:09:16 +0900] [26697] [INFO] Listening at: http://127.0.0.1:8000 (26697)
[2019-05-20 23:09:16 +0900] [26697] [INFO] Using worker: sync
[2019-05-20 23:09:16 +0900] [26700] [INFO] Booting worker with pid: 26700

```

### CODE REVIEW

File: gunicorn/util.py [hunk 1]
```diff
@@ -364,15 +369,17 @@ def import_app(module):
...
 
     try:
-        __import__(module)
-    except ImportError:
+        if importlib:
+            mod = importlib.import_module(module)
+        else:
+            __import__(module)
+            mod = sys.modules[module]
+    except (ImportError, ModuleNotFoundError):
```
Comment (c-bata) [inline gunicorn/util.py]:
```
$ python -V
Python 3.7.2
$ gunicorn examples/echo.py:app
[2019-05-20 23:06:27 +0900] [26485] [INFO] Starting gunicorn 19.9.0
[2019-05-20 23:06:27 +0900] [26485] [INFO] Listening at: http://127.0.0.1:8000 (26485)
[2019-05-20 23:06:27 +0900] [26485] [INFO] Using worker: sync
[2019-05-20 23:06:27 +0900] [26488] [INFO] Booting worker with pid: 26488
[2019-05-20 23:06:27 +0900] [26488] [ERROR] Exception in worker process
Traceback (most recent call last):
  File "/Users/c-bata/src/github.com/benoitc/gunicorn/gunicorn/util.py", line 373, in import_app
    mod = importlib.import_module(module)
  File "/usr/local/Cellar/python/3.7.2_2/Frameworks/Python.framework/Versions/3.7/lib/python3.7/importlib/__init__.py", line 127, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 1006, in _gcd_import
  File "<frozen importlib._bootstrap>", line 983, in _find_and_load
  File "<frozen importlib._bootstrap>", line 953, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "<frozen importlib._bootstrap>", line 1006, in _gcd_import
  File "<frozen importlib._bootstrap>", line 983, in _find_and_load
  File "<frozen importlib._bootstrap>", line 965, in _find_and_load_unlocked
ModuleNotFoundError: No module named 'examples/echo'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/Users/c-bata/src/github.com/benoitc/gunicorn/gunicorn/arbiter.py", line 585, in spawn_worker
    worker.init_process()
  File "/Users/c-bata/src/github.com/benoitc/gunicorn/gunicorn/workers/base.py", line 133, in init_process
    self.load_wsgi()
  File "/Users/c-bata/src/github.com/benoitc/gunicorn/gunicorn/workers/base.py", line 142, in load_wsgi
    self.wsgi = self.app.wsgi()
  File "/Users/c-bata/src/github.com/benoitc/gunicorn/gunicorn/app/base.py", line 65, in wsgi
    self.callable = self.load()
  File "/Users/c-bata/src/github.com/benoitc/gunicorn/gunicorn/app/wsgiapp.py", line 49, in load
    return self.load_wsgiapp()
  File "/Users/c-bata/src/github.com/benoitc/gunicorn/gunicorn/app/wsgiapp.py", line 39, in load_wsgiapp
    return util.import_app(self.app_uri)
  File "/Users/c-bata/src/github.com/benoitc/gunicorn/gunicorn/util.py", line 380, in import_app
    raise ImportError(msg % (module.rsplit(".", 1)[0], obj))
ImportError: Failed to find application, did you mean 'examples/echo:app'?
[2019-05-20 23:06:27 +0900] [26488] [INFO] Worker exiting (pid: 26488)
[2019-05-20 23:06:27 +0900] [26485] [INFO] Shutting down: Master
[2019-05-20 23:06:27 +0900] [26485] [INFO] Reason: Worker failed to boot.
```

File: gunicorn/util.py [hunk 2]
```diff
@@ -364,15 +369,17 @@ def import_app(module):
         module, obj = parts[0], parts[1]
 
     try:
-        __import__(module)
-    except ImportError:
+        if importlib:
+            mod = importlib.import_module(module)
```
Comment (c-bata) [inline gunicorn/util.py]:
```
$ python -V
Python 3.7.2
$ gunicorn examples.echo:app
[2019-05-20 23:09:16 +0900] [26697] [INFO] Starting gunicorn 19.9.0
[2019-05-20 23:09:16 +0900] [26697] [INFO] Listening at: http://127.0.0.1:8000 (26697)
[2019-05-20 23:09:16 +0900] [26697] [INFO] Using worker: sync
[2019-05-20 23:09:16 +0900] [26700] [INFO] Booting worker with pid: 26700

```

File: gunicorn/util.py [hunk 3]
```diff
@@ -364,19 +365,17 @@ def import_app(module):
...
         raise
 
-    mod = sys.modules[module]
-
     is_debug = logging.root.level == logging.DEBUG
     try:
-        app = eval(obj, vars(mod))
-    except NameError:
+        app = getattr(mod, obj)
+    except AttributeError:
```
Comment (c-bata) [inline gunicorn/util.py]: Error message when failing to find application object.

```
$ gunicorn examples.echo:foo
[2019-05-20 23:19:04 +0900] [27282] [INFO] Starting gunicorn 19.9.0
[2019-05-20 23:19:04 +0900] [27282] [INFO] Listening at: http://127.0.0.1:8000 (27282)
[2019-05-20 23:19:04 +0900] [27282] [INFO] Using worker: sync
[2019-05-20 23:19:04 +0900] [27285] [INFO] Booting worker with pid: 27285
Failed to find application object 'foo' in 'examples.echo'
[2019-05-20 23:19:04 +0900] [27285] [INFO] Worker exiting (pid: 27285)
[2019-05-20 23:19:05 +0900] [27282] [INFO] Shutting down: Master
[2019-05-20 23:19:05 +0900] [27282] [INFO] Reason: App failed to load.
```

File: gunicorn/util.py [hunk 4]
```diff
@@ -364,19 +365,18 @@ def import_app(module):
         module, obj = parts[0], parts[1]
 
     try:
-        __import__(module)
+        mod = importlib.import_module(module)
     except ImportError:
+        # ModuleNotFoundError (a subclass of ImportError) is new in version 3.6.
```
Comment (berkerpeksag) [inline gunicorn/util.py]: This comment can be dropped.

Comment (c-bata) [reply gunicorn/util.py]: I just wanted to tell `ModuleNotFound` is handled by this clause. But It's ok to remove this comment.

Comment (tilgovi) [reply gunicorn/util.py]: It's probably not necessary to have the comment. There's only one statement in the `try` clause, so even if the error is more general than needed on Python 3.6 there's not much need to justify it.

Comment (c-bata) [reply gunicorn/util.py]: I see. I've just removed this line.

**Comment by berkerpeksag** [comment]
Nice cleanup, thanks!

**Pull request merged**